ui/packages/components/src/types/workers.ts

```diff
@@ -39,7 +39,7 @@ export type Worker = {
   functionCount: number;
 };
 
-export const groupedWorkerStatuses = ['ACTIVE', 'INACTIVE', 'FAILED'] as const;
+export const groupedWorkerStatuses = ['ACTIVE', 'INACTIVE', 'FAILED', 'DISCONNECTED'] as const;
```

Comment:
In your current implementation 'FAILED' doesn't exist anymore

Reply:
```
enum ConnectionStatus {
	CONNECTED = 0;
	READY = 1;
	DRAINING = 2;
	DISCONNECTING = 3;
	DISCONNECTED = 4;
}
```

These are the connection statuses used at the moment. We don't have a separate failed status, so we could remove that from the UI for the moment.

Reply:
We do expose a `disconnectReason` which we could display, adding more information.